.github/workflows/spice_postgres_bench_docker.yml

```diff
@@ -0,0 +1,62 @@
+name: spice-postgres-bench docker
+
+on:
+  push:
+    branches:
+      - qianqian/docker-tpch-postgres
+  pull_request:
+    branches:
+      - qianqian/docker-tpch-postgres
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Pull Postgres Docker image
+        run: docker pull postgres:latest
+
+      - name: Start temporary Postgres server
+        run: |
+          docker run --name spice-postgres-bench-temp -e POSTGRES_PASSWORD=postgres -p 5432:5432 -d postgres:latest
+          sleep 10
+
+      - name: Install necessary tools
+        run: sudo apt-get update && sudo apt-get install -y git make gcc
+
+      - name: Initialize TPCH data
+        working-directory: ./test/tpch
+        run: |
+          make tpch-init
+          export PGPASSWORD='postgres'
+          for sf in 0.01 0.05 1; do
+            DB_NAME=tpch_sf${sf//./_}
+            DBGEN_SCALE=$sf make tpch-gen
+            DB_HOST=127.0.0.1 DB_PORT=5432 DB_USER=postgres DB_NAME=$DB_NAME make pg-init
```

Comment:
This is starting a Postgres server, generating the TPCH data and then dumping the TPCH data from Postgres. Is there a way to just generate the SQL directly without having to stage it in Postgres first?

Reply:
To create a Postgres Docker image that has pre-populated database would require a SQL dump that contains all the commands of populating data. To get these SQL of populating data would require a staging database popultaed data, and use pg_dump over it.